src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -115,20 +113,9 @@ class ProtobufPlugin implements Plugin<Project> {
         }
     }
 
-    private static List<String> getLanguages(Project project) {
-      List<String> additionalLanguages = []
-      if (project.hasProperty(USER_LANG_PROP)) {
-        additionalLanguages = (List<String>) project.property(USER_LANG_PROP)
-        project.logger.log(
-            LogLevel.WARN,
-            "protobuf plugin is now using additional unsupported languages: " + additionalLanguages)
-      }
-      return SUPPORTED_LANGUAGES + additionalLanguages
-    }
-
     private static void linkGenerateProtoTasksToTask(Task task, GenerateProtoTask genProtoTask) {
       task.dependsOn(genProtoTask)
-      task.source genProtoTask.getOutputSourceDirectorySet()
+      task.source genProtoTask.getOutputSourceDirectorySet().include("**/*.java", "**/*.kt")
```

Comment:
How does this fix the "no source files" error?

Reply:
Applying the Java/Kotlin filter can avoid feeding non-Java/Kotlin source files to Java compilation task, which avoids task input change that triggers the task to run (if there are no Java/Kotlin source files, Java compilation tasks should not run).

Reply:
We put each language in its own folder. Why don't we put each in its own SourceDirectorySet?

Reply:
Each SourceSet can have multiple compile tasks as for different languages (see SourceSet's [`getComileTaskName(lang)`](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceSet.html#getCompileTaskName-java.lang.String-) API). However, each SourceSet has only one `GenerateProtoTask`, which generates code for all languages specified. So for each SourceSet, we are hooking the output of its `GenerateProtoTask` as the input to each of its compile tasks (for each language). The output of a `GenerateProtoTask` is a [single `SourceDirectorySet`](https://github.com/google/protobuf-gradle-plugin/blob/c425e558f57764b791b103614b5fada7d05c41da/src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy#L504) that contains generated code for plugin/builtin in separated directories. We are not able to map plugins/builtins to individual languages as they do not contain language information.